java/build/bazel/tests/integration/WorkspaceDriver.java

```diff
@@ -155,6 +151,27 @@ protected void copyFromRunfiles(String path, String destpath) throws IOException
     Files.copy(origin.toPath(), dest.toPath());
   }
 
+  /**
+   * Copy the whole directory from the runfiles under {@code rootDirectoryPath} to the current workspace.
+   */
+  protected void copyFromRunfilesDirectory(final String rootDirectoryPath) throws IOException {
+    File root = getRunfile(rootDirectoryPath);
+    try (Stream<Path> paths = Files.walk(Paths.get(root.toURI()))) {
+      paths.filter(path -> new File(path.toUri()).isFile())
+              .forEach(file -> {
+                String relativeToRunfiles = file.toAbsolutePath().toString().substring(runfileDirectory.getPath().length());
+                Path relativeToRunfilesPath = Paths.get(relativeToRunfiles);
+                String destination = relativeToRunfilesPath.subpath(1, relativeToRunfilesPath.getNameCount()).toString();
```

Comment:
wdyt about doing a private method here? for readability. This is probably something like "strip first directory". First is meh name but root is also problematic since it's root in relation to the runfiles